x/interchainstaking/keeper/msg_server.go

```diff
@@ -122,12 +130,112 @@ func (k msgServer) CancelRedemption(goCtx context.Context, msg *types.MsgCancelQ
 		sdk.NewEvent(
 			types.EventTypeRedemptionCancellation,
 			sdk.NewAttribute(types.AttributeKeyReturnedAmount, record.BurnAmount.String()),
-			sdk.NewAttribute(types.AttributeKeyUser, msg.FromAddress),
+			sdk.NewAttribute(types.AttributeKeyUser, record.Delegator),
+			sdk.NewAttribute(types.AttributeKeyHash, msg.Hash),
+			sdk.NewAttribute(types.AttributeKeyChainID, msg.ChainId),
+		),
+	})
+
+	return &types.MsgCancelRedemptionResponse{Returned: record.BurnAmount}, nil
+}
+
+func (k msgServer) RequeueRedemption(goCtx context.Context, msg *types.MsgRequeueRedemption) (*types.MsgRequeueRedemptionResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check for errored unbond in UNBONDING status
+	record, found := k.GetWithdrawalRecord(ctx, msg.ChainId, msg.Hash, types.WithdrawStatusUnbond)
+	if !found {
+		return nil, fmt.Errorf("no unbonding record with hash \"%s\" found", msg.Hash)
+	}
+	if record.SendErrors == 0 {
+		return nil, fmt.Errorf("cannot requeue unbond \"%s\" with no errors", msg.Hash)
+	}
+
+	if record.Delegator != msg.FromAddress && k.Keeper.GetGovAuthority(ctx) != msg.FromAddress {
+		return nil, fmt.Errorf("incorrect user for record with hash \"%s\"", msg.Hash)
```

Comment:
In the future @joe-bowman @faddat we can simply use the format specifier "%q" per https://go.dev/play/p/7SJTkPPMyFl